src/lib/tank.js

```diff
@@ -54,67 +54,61 @@ const ensureFundsFor = async (
+  const balance = toBN(await web3.eth.getBalance(address));
   cost = toBN(cost);
 
-  if (cost.gt(balance)) {
-    try {
-      if (point !== null) {
-        const fundsRemaining = await remainingTransactions(point);
-        if (fundsRemaining < signedTxs.length) {
-          throw new Error('tank: request invalid');
-        }
-      } else {
-        console.log('tank: skipping remaining-funds check');
-        //TODO if we can't always (easily) provide a point, and the
-        //     fundTransactions call is gonna fail anyway, should we maybe
-        //     just not bother doing this check in the first place?
-      }
+  if (balance.gte(cost)) {
+    console.log(
+      `tank: already have sufficient funds: ${address} proposed ` +
+        `a transacton(s) that will cost ${cost.toString()}wei but it ` +
+        `already has ${balance.toString}wei`
+    );
+    return false;
+  }
 
-      const res = await fundTransactions(signedTxs);
-      if (!res.success) {
-        throw new Error('tank: request rejected', res);
-      } else {
-        await waitForTransactionConfirm(web3, res.txHash);
-        let newBalance = await web3.eth.getBalance(address);
-        console.log(
-          'tank: funds have confirmed',
-          balance.toString(),
-          newBalance
-        );
-        return true;
+  try {
+    // TODO: if we can't always (easily) provide a point, and the
+    // fundTransactions call is gonna fail anyway, should we maybe
+    // just not bother doing this check in the first place?
+    if (point !== null) {
+      const fundsRemaining = await remainingTransactions(point);
+      if (fundsRemaining < signedTxs.length) {
+        throw new Error('tank: request invalid');
       }
-    } catch (e) {
-      console.log('tank: funding failed', e);
-      await waitForBalance(
-        web3,
-        address,
-        cost.toString(),
-        askForFunding,
-        gotFunding
-      );
+    } else {
+      console.log('tank: no point provided. skipping remaining-funds check...');
     }
-  } else {
+
+    const res = await fundTransactions(signedTxs);
+    if (!res.success) {
+      throw new Error('tank: request rejected', res);
+    }
+
+    await waitForTransactionConfirm(web3, res.txHash);
+
+    const newBalance = await web3.eth.getBalance(address);
     console.log(
-      'tank: already have sufficient funds',
-      cost.toString(),
-      address,
-      balance.toString()
+      `tank: funds have confirmed: ${address} now has ` +
+        ` ${newBalance.toString()}wei, up from ${balance.toString()}wei`
     );
+    return true;
+  } catch (e) {
+    console.log('tank: funding failed, waiting on user funds...', e);
+    await waitForBalance(web3, address, cost, askForFunding, gotFunding);
   }
+
   return false;
 };
 
 // returns a promise that resolves when address has at least minBalance
 function waitForBalance(web3, address, minBalance, askForFunding, gotFunding) {
-  minBalance = toBN(minBalance);
```

Comment:
I put this here to ensure that `minBalace` is always a BN, so we don't run into issues with the `.gte()`. Pretty sure `toBN` on a BN is a cheap no-op. Is there a specific reason you want to take this out?

Reply:
I actually pass minBalance as a BN here, since this function is only used internally, but I realize that's not conforming to the contract where we input and output strings. will change it back